tests/mpls_test.py

```diff
@@ -23,71 +24,201 @@
 
+class TestMplsMasic(object):
+    @classmethod
+    def setup_class(cls):
+        print("SETUP")
+        from .mock_tables import mock_multi_asic
+        importlib.reload(mock_multi_asic)
+        from .mock_tables import dbconnector
+        dbconnector.load_namespace_config() 
+
+    def test_config_mpls_masic_add(self):
+        runner = CliRunner()
+        db = Db()
+        obj = {'config_db':db.cfgdb, 'namespace':'asic0'}
+
+        result = runner.invoke(config.config.commands["interface"].commands["mpls"].commands["add"], ["Ethernet8"], obj=obj)
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        assert db.cfgdb.get_entry("INTERFACE", "Ethernet8") == {"mpls": "enable"}
+
+
+    def test_show_interfaces_mpls_masic_frontend(self):
         runner = CliRunner()
         result = runner.invoke(show.cli.commands["interfaces"].commands["mpls"], [])
         print(result.exit_code)
         print(result.output)
         assert result.exit_code == 0
-        assert result.output == show_interfaces_mpls_output
+        assert result.output == show_interfaces_mpls_masic_output_frontend
+
+    def test_show_interfaces_mpls_masic_all(self):
+        runner = CliRunner()
+        result = runner.invoke(show.cli.commands["interfaces"].commands["mpls"], ["-dall"])
+        print(result.exit_code)
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_interfaces_mpls_masic_output_all
 
-    def test_show_interfaces_mpls_specific(self):
-        jsonfile = os.path.join(mock_db_path, 'appl_db')
-        dbconnector.dedicated_dbs['APPL_DB'] = jsonfile
+    def test_show_interfaces_mpls_masic_asic(self):
+        runner = CliRunner()
+        result = runner.invoke(show.cli.commands["interfaces"].commands["mpls"], ["-nasic0"])
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_interfaces_mpls_masic_output_frontend
 
+    def test_show_interfaces_mpls_masic_asic_all(self):
         runner = CliRunner()
-        result = runner.invoke(show.cli.commands["interfaces"].commands["mpls"], ["Ethernet2"])
+        result = runner.invoke(show.cli.commands["interfaces"].commands["mpls"], ["-nasic0", "-dall"])
+        print(result.output)
+        assert result.exit_code == 0
+        assert result.output == show_interfaces_mpls_masic_output_asic_all
+    
+    def test_config_mpls_masic_remove(self):
+        runner = CliRunner()
+        db = Db()
+        obj = {'config_db':db.cfgdb, 'namespace':'asic0'}
+
+        result = runner.invoke(config.config.commands["interface"].commands["mpls"].commands["remove"], ["Ethernet8"], obj=obj)
         print(result.exit_code)
         print(result.output)
         assert result.exit_code == 0
-        assert result.output == show_interfaces_mpls_specific_output
+        assert db.cfgdb.get_entry("INTERFACE", "Ethernet8") == {"mpls": "disable"}
 
```

Comment:
How about adding a negative test case for namespace. 'i.e' pass an invalid namespace in the option "-ndummy". This way  you can test "multi_asic.validation" callback.

Reply:
unfortunately not possible to test, as variables cannot be modified